src/scripts/glue_jobs/glue_job.py

```diff
@@ -73,30 +74,32 @@ def execute(self):
 
     # Resolve the data type mismatch by comparing it with table schema and store the processed data back to s3
     def resolve_choice_and_write_output_frame(self, transformed_frame, partition_key):
-        job_name = self.args["JOB_NAME"]
-        if transformed_frame.stageErrorsCount() > 0:
-            self.log_errors(transformed_frame)
+        self.check_and_log_errors(transformed_frame)
 
         output_frame = transformed_frame.resolveChoice(choice='match_catalog', database=self.args['DATABASE_NAME'],
                                                        table_name=self.args['TABLE_NAME'])
-        self.logger.info("Total output records write to s3 for job {} is {}".format(job_name,
-                                                                                    str(output_frame.count())))
+        self.check_and_log_errors(output_frame)
+
         if output_frame.count() > 0:
+            self.logger.info("The total number of records saved to s3 is {}".format(output_frame.count()))
             self.glue_context.write_dynamic_frame.from_catalog(
                 frame=output_frame,
                 database=self.args["DATABASE_NAME"],
                 table_name=self.args["TABLE_NAME"],
                 additional_options={"partitionKeys": [partition_key]}
             )
         self.job.commit()
-        self.logger.info("Glue job {} finished.".format(self.args["JOB_NAME"]))
+        self.logger.info("Glue job finished.")
 
-    def log_errors(self, dynamic_frame):
-        job_name = self.args["JOB_NAME"]
-        self.logger.info("Error count is {} in job {}".format(str(dynamic_frame.stageErrorsCount()),
-                                                              job_name))
-        error_record = dynamic_frame.errorsAsDynamicFrame().toDF().head()
-        error_fields = error_record["error"]
-        for key in error_fields.asDict().keys():
-            self.logger.info("\n Job {} has error {} : {}".format(job_name, key, str(error_fields[key])))
-        raise Exception("Error in job! See the log!")
+    def check_and_log_errors(self, dynamic_frame):
+        if dynamic_frame.stageErrorsCount() > 0:
+            self.logger.error(
+                "The total number of error count in dynamic frame is {} ".format(dynamic_frame.stageErrorsCount()))
+            df = dynamic_frame.errorsAsDynamicFrame().toDF()
+            last_error_msg = "UNKNOWN"
+            for row in df.rdd.collect():
```

Comment:
See https://sparkbyexamples.com/pyspark/pyspark-loop-iterate-through-rows-in-dataframe/